frontend/packages/dev-console/integration-tests/utilities/elementInteractions.ts

```diff
@@ -0,0 +1,91 @@
+    await browser.wait(EC.visibilityOf(drp_ele), timeoutInMilliseconds, `${drp_ele} is not able to click in DOM, even after ${timeoutInMilliseconds} milliseconds`);
+    await drp_ele.click();
+    await browser.wait(EC.elementToBeClickable(element(by.css('ul.pf-c-dropdown__menu'))), timeoutInMilliseconds);
+    await element(by.cssContainingText('li button.pf-c-dropdown__menu-item', drpdownListvalue)).click();
+  }
+
+  // Perform mouseHover on the given element and perform click action with in given time
+  export async function mouseHoverClick(ele: any, timeoutInMilliseconds = waitForElement) {
+    await browser.executeScript('arguments[0].scrollIntoView();', ele);
+    await browser.wait(EC.presenceOf(ele), timeoutInMilliseconds, `${ele} is not able to click in DOM, even after ${timeoutInMilliseconds} milliseconds`);
+    await browser.actions().mouseMove(ele).click().perform();
+  }
+
+  // Select the dropdown value from given dropdown element with in given time
+  export async function selectByVisibleText(ele: any, drpdownListvalue: string, timeoutInMilliseconds = waitForElement) {
+    await browser.wait(EC.visibilityOf(ele), timeoutInMilliseconds, `${ele} is not able to click in DOM, even after ${timeoutInMilliseconds} milliseconds`);
+    await browser.executeScript('arguments[0].scrollIntoView();', ele);
+    await ele.click();
+    await browser.wait(EC.elementToBeClickable(element(by.css('ul.pf-c-dropdown__menu'))), timeoutInMilliseconds);
+    await element(by.cssContainingText('li button.pf-c-dropdown__menu-item', drpdownListvalue)).click();
+  }
+
+  /* Select the dropdown value based on the index for given dropdown element with in given time. If index is not provided by default 0th index is considered */
+  export async function selectByIndex(ele: any, index: number = 0, timeoutInMilliseconds = waitForElement) {
+    await browser.wait(EC.elementToBeClickable(ele), timeoutInMilliseconds, `${ele} is not able to click in DOM, even after ${timeoutInMilliseconds} milliseconds`);
+    await browser.executeScript('arguments[0].scrollIntoView();', ele);
+    await ele.click();
+    await browser.wait(EC.elementToBeClickable(element(by.css('ul.pf-c-dropdown__menu')), timeoutInMilliseconds, `Unable to view the dropdown options even after ${timeoutInMilliseconds} ms`));
+    await element.all(by.css('li button.pf-c-dropdown__menu-item')).get(index).click();
+  }
+
+  /* Select the checkbox with in given time */
+  export async function selectCheckBox(ele: any, timeoutInMilliseconds = waitForElement) {
+    await browser.wait(EC.elementToBeClickable(ele), timeoutInMilliseconds, `${ele} is not able to click in DOM, even after ${timeoutInMilliseconds} milliseconds`);
+    await browser.executeScript('arguments[0].scrollIntoView();', ele);
+    let result = await verifyCheckBox(ele)
+    if(result == false) {
+      await ele.click(); 
+    } else {
+      console.log('Check Box is already selected');
```

Comment:
This is curious... why have this as part of test utilities? Is this for debugging purposes? Do we have another way to showcase this?

Reply:
Yeah, I included this console statement for debugging purpose, But soon I am planning to bring logger statements which stores the logs in separate files. But for now I will delete this else block which doesn't have any impact